compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/validator/FunctionValidator.java

```diff
@@ -306,11 +322,24 @@ private void addInputTypeErrorIntoContext(TypeSymbol typeSymbol, Location inputL
             Symbol symbol = ((TypeReferenceTypeSymbol) typeSymbol).definition();
             if (symbol.kind() == SymbolKind.CLASS) {
                 updateContext(context, CompilationError.INVALID_RESOURCE_INPUT_PARAM, inputLocation);
+            } else if (isFileUploadParameter(typeSymbol)) {
+                updateContext(context, CompilationError.INVALID_LOCATION_FOR_FILE_UPLOAD_PARAMETER, inputLocation);
             } else {
                 TypeDefinitionSymbol typeDefinitionSymbol = (TypeDefinitionSymbol) symbol;
                 if (typeDefinitionSymbol.typeDescriptor().typeKind() == TypeDescKind.RECORD) {
-                    updateContext(context, CompilationError.INVALID_RESOURCE_INPUT_OBJECT_PARAM,
-                            inputLocation);
+                    boolean fieldError = false;
+                    Map<String, RecordFieldSymbol> memberMap = ((RecordTypeSymbol) typeDefinitionSymbol
+                            .typeDescriptor()).fieldDescriptors();
+                    for (RecordFieldSymbol fields : memberMap.values()) {
+                        if (isFileUploadParameter(fields.typeDescriptor())) {
+                            fieldError = true;
+                            updateContext(context, CompilationError.INVALID_INPUT_OBJECT_FIELD_TYPE, inputLocation);
+                        }
+                    }
+                    if (!fieldError) {
+                        updateContext(context, CompilationError.INVALID_RESOURCE_INPUT_OBJECT_PARAM,
+                                inputLocation);
+                    }
```

Comment:
Aren't we already validate the record fields? What's the purpose of validating the record fields once again here?